### PR TITLE
[digitalstrom] Adoptions due to API changes in DSS Update 1.19.2

### LINKED ISSUE #11847: [digitalstrom] Power Consumption and Energy Level NULL after DSS Update to 1.19.2
Digitalstrom did an Update to dss Version 1.19.2. After I installed the update, all my power and energy values get no updates and if I add a new Item, the value is NULL.

It seems, that digitalstrom changed something here, so we do not get values anymore.

In the [Release notes](https://www.digitalstrom.com/wp-content/uploads/2021/12/SRN-2112_v1.1_EN.pdf) is written, that they improved something with the energy graphics. 

I am using the openhab 3.2.0 docker image. Problem was in 3.1.1, too. I restarted the dss and also openhab was restarted a couple of times.

For me it seems that we have a new bug here.

To reproduce that Bug, you have to update the digitalstrom server to version 1.19.2. After that you will not see any item value updates on your power consumptions or energy levels. So it just keeps at 0. Interesting is, that the totalpowerconsumption, which belongs to the digitalstrom server thing, is still working.

I am using a Debian 10 Installation on a Raspberry Pi 4 with openhab docker image in version 3.2.0. The Digitalstrom Server is a dss20 with distroVersion 1.19.2

### PR DESCRIPTION
**Pull request by alexf2015**
Fixes #11847
This should fix the issue introduced by DSS update to 1.19.2 while staying backwards compatible to older versions of DSS.

### CODE REVIEW

**Comment by lolodomo** [comment]
Please apply spotless: `mvn spotless:apply`

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    private String dsuid;
+    private final String DEFAULT_DSUID = "3504175fe0000000000000000000000001";
+    private final String PRE = "3504175fe0000000";
+    private final String ALL = "ALL";
+
+    /**
+     * Creates a new {@link DSUID}.
+     *
+     * @param dsuid to create
```
Comment (lolodomo) [inline bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java:30]: Parameter is in fact dsid. 
But it should probably be suïd like mentioned here?

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    public DSUID(String dsid) {
+        this.dsuid = dsid;
+        if (dsid != null && !dsid.trim().equals("")) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java:34]: You could put the result of dsid.trim() in a variable as you called it several times.

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            }
+        } else {
+            this.dsuid = DEFAULT_DSUID;
+        }
+    }
+
+    /**
+     * Returns the dSUID as {@link String}.
+     *
+     * @return dSID
```
Comment (lolodomo) [inline bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java:52]: DSUID

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by hmerk** [comment]
@lolodomo We need to remove MichaelOechel and msiegele from codeowners, as they have finished their student project and will no onger maintaingin the binding.

**Comment by lolodomo** [comment]
How could replace them? 
Searching who contributed for this binding in 2021, I see @ardanedh as the most regular contributor. 

@ardanedh @alexf2015 : is one of you knowing well the code of this binding. If this is case, would like to review future PRs?

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java (hunk 1 above)
Comment (alexf2015) [reply bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java:30]: c&p issue. I also removed some of those if cases as I actually don't know if that could occour, e.g. a short version of dsuid which must be expanded. Looks strange as the expansion string is a fixed string.

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java (hunk 2 above)
Comment (alexf2015) [reply bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSUID.java:34]: just copied from DSID class, now simplified a lot.

**Comment by ardanedh** [comment]
> Who could replace them? Searching who contributed for this binding in 2021, I see @ardanedh as the most regular contributor.
> 
> @ardanedh @alexf2015 : is one of you knowing well the code of this binding. If this is the case, would you like to review future PRs?

The codebase of the binding is huge - I've been rummaging around a bit and as @alexf2015 already mentioned, the original intent is difficult to grasp. In my opinion, there are too many digitalSTROM concepts realised in the binding - and that's exactly what I don't need when using openHAB.

I'm actively using the binding and I am willing to help keep it running. But after all, this is an open source project and no one can be forced to maintain business service level reaction times ;)

Obviously, digitalSTROM is working on a new [API](https://developer.digitalstrom.org/api/#overview--public-api), which would justify a reimplementation of a (lean, DS concept free) digitalSTROM binding ...

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@ardanedh ; can you at least tell us if this PR looks good to you?

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSID.java [hunk 4]
```diff
@@ -22,24 +22,21 @@
...
-            if (dsid.trim().length() == 24) {
-                this.dsid = dsid;
-            } else if (dsid.trim().length() == 8) {
-                this.dsid = this.PRE + dsid;
-            } else if (dsid.trim().toUpperCase().equals("ALL")) {
-                this.dsid = "ALL";
-            } else {
-                this.dsid = DEFAULT_DSID;
-            }
+        var trimmedDsuid = dsid != null ? dsid.trim() : "";
```
Comment (ardanedh) [inline bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSID.java:33]: Should be named `trimmedDsid`

**Review by ardanedh — APPROVED** [review]
Apart from the `trimmedDsuid`, this looks fine to me as well ... thank you, @alexf2015!

File: bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSID.java (hunk 4 above)
Comment (alexf2015) [reply bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/structure/devices/deviceparameters/impl/DSID.java]: done

**Review by ardanedh — APPROVED** [review]

**Pull request merged**